[Feature] 테스트용 ClickHouse GKE 내부 배포

cbbsjj0314 opened the pull request:
## 📌 테스트용 ClickHouse GKE 내부 배포

## ✨ 변경 사항  
- GKE 테스트 환경에서 실시간 데이터 저장 및 압축률 실험을 위한 ClickHouse 단일 노드 배포 설정 추가
- Altinity Helm Chart를 사용하며, Helm 배포 로직을 `deploy_stacks.sh`에 통합

## 📂 변경된 파일  
- `deploy/clickhouse/clickhouse-values.yaml`: ClickHouse Helm 배포 설정 파일
- `deploy/clickhouse/clickhouse_deploy.sh`: ClickHouse Helm 배포 스크립트 추가

## 🚀 변경 유형  
- [x] ✨ 기능 추가 (새로운 기능)  
- [ ] 🐛 버그 수정 (기존 기능 수정)  
- [ ] ♻️ 리팩토링 (코드 구조 개선)  
- [ ] 📝 문서 수정 (README 등 문서 변경)  
- [ ] 🚨 테스트 추가/수정  

## ✅ 체크리스트  
PR을 올리기 전에 아래 사항을 확인해주세요.  
- [x] 코드가 정상적으로 실행됨  
- [ ] 관련 테스트 코드를 추가함  
- [x] 기존 기능에 영향을 주지 않음  
- [ ] 문서를 업데이트했음 (필요한 경우)  

## 💬 기타 참고 사항  
- ClickHouse 비밀번호는 보안을 고려해 Git에 포함하지 않아, 수동 생성 필요 (비밀 채널에 공유)
- 해당 시크릿이 존재하지 않으면 `deploy_stacks.sh` 실행이 중단되도록 안전장치 포함
- lickHouse는 내부 GKE 통신을 위한 ClusterIP 방식으로 배포됨

cbbsjj0314 committed:
feat: add ClickHouse Helm values for single-node GKE test deployment
- Added deploy/clickhouse/clickhouse-values.yaml
- Configured for Altinity-based single-node setup
- keeper disabled for standalone mode

cbbsjj0314 committed:
feat: add ClickHouse deployment step to deploy_stacks.sh
- Deploys Altinity ClickHouse via Helm in test-clickhouse namespace
- Includes namespace setup and values.yaml reference
- Follows same structure as other components (Airflow, Spark, ELK)
- Requires pre-created Kubernetes Secret `test-clickhouse-password` with ClickHouse password

CoffeeIsNak commented on deploy/clickhouse/clickhouse-values.yaml:
```diff
@@ -0,0 +1,50 @@
+clickhouse:
+  configurationFiles:
+    config.d:
+      # Zookeeper 설정 제거 (단일 노드 운영 시 필요 x)
```
실제로 저희 사용될 클릭하우스가 단일 노드면 적절할까용?!

cbbsjj0314 replied:
저번 회의 때 일단은 한 번 합치자고 해서 이대로 올리긴 했습니다만,
2~3개 클러스터로 구성하도록 변경할 예정임다

CoffeeIsNak commented on deploy/clickhouse/clickhouse-values.yaml:
```diff
@@ -0,0 +1,50 @@
+  volumeClaimTemplates:
+    - name: data
+      accessMode: ReadWriteOnce
+      storageClassName: standard
```
standard가 아마 HDD인걸로 파악하고 있는데, 괜찮으신걸로 생각하면 될까요?! 
GKE 노드 같은 경우 HDD를 쓰고 있고 HDD 역시 리전별 제한이 500기가로 알고 있습니다! 

만약 다른 분들의 작업 편의를 고려해주셔서 양보해주신 것이라면, SSD 용량을 사용해주셔도 괜찮다고 생각되옵니다!

cbbsjj0314 replied:
오 그럼 SSD로 바꾸겠슴다

boolYikes replied:
fuse타입은 gcs를 마운트할수도 있나보네요 근데 머가 더 싼지는 모르겟어요 🤔

CoffeeIsNak reviewed (commented):
좋은 PR 감사합니다. 리뷰 코멘트를 몇개 남겼습니다 ~ 확인 부탁드립니다!

cbbsjj0314 committed:
feat: extract ClickHouse deployment into separate script

cbbsjj0314 committed:
chore: adjust CPU/memory and switch storage to SSD

- Changed storageClass from 'standard' (HDD) to 'balanced' (GCE SSD)
- and increased the volume size to 100Gi to ensure sufficient storage capacity
